Slim/Networking/Async.pm

```diff
@@ -167,7 +167,7 @@ sub _async_connect {
 
 	# check that we are actually connected
 	if ( !$socket->connected ) {
-		if ( ref $socket eq 'Slim::Networking::Async::Socket::HTTPS' and $! == EWOULDBLOCK ) {
+		if ($socket->isa('Slim::Networking::Async::Socket::HTTPS') && ($! == EWOULDBLOCK || $! == EAGAIN)) {
```

Comment:
Are you saying that the important part here is the use of `EAGAIN`, rather than the class check?

Reply:
That said, what you're saying about the difference between `isa` and `ref` makes sense.

Reply:
yes, the EAGAIN was the issue, if Had no real reason to make the isa change :-). I can't remember as I'm writing that but I know I've been having these issues between Linux and Windows platform of EWOULDBLOCK and/or EAGAIN, so usually I endup using both. I think they point to the same value on many platform, but it seems not in our case